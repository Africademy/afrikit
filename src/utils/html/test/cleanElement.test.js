// // @flow

// import expect from 'expect';
// import React from 'react';
// import styled from 'styled-components';
// import { renderToString } from 'react-dom/server';
// import { textAlign, styles } from '../../../theme/space';
// import cleanElement from '../cleanElement';

// const { propTypes } = styles;

// describe('cleanElement', () => {
//   it('removes props with prop types', () => {
//     const Clean = cleanElement('div');
//     Clean.propTypes = {
//       ...propTypes.textAlign,
//     };
//     const Box = styled(Clean)`
//       ${textAlign};
//     `;

//     const result = renderToString(
//       <Box id="beep" textAlign="center">
//         hello
//       </Box>,
//     );

//     // t.snapshot(result);
//     expect(result).toMatchSnapshot();
//   });

//   it('does not remove props without propTypes', () => {
//     const Clean = cleanElement('div');
//     const Box = styled(Clean)`
//       ${textAlign};
//     `;

//     const result = renderToString(
//       <Box id="beep" textAlign="center">
//         hello
//       </Box>,
//     );

//     // t.snapshot(result);
//     expect(result).toMatchSnapshot();
//   });
// });
