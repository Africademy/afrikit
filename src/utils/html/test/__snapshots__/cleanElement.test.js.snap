// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cleanElement does not remove props without propTypes 1`] = `"<div class=\\"cleanElementtest__Box-kdszWu eYbyRo\\" id=\\"beep\\" textAlign=\\"center\\" data-reactroot=\\"\\">hello</div>"`;

exports[`cleanElement removes props with prop types 1`] = `"<div class=\\"cleanElementtest__Box-kdszWu eYbyRo\\" id=\\"beep\\" data-reactroot=\\"\\">hello</div>"`;
